src/app/components/table/table.component.ts

```diff
@@ -1,20 +1,26 @@
-import { Component, OnInit, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, OnDestroy } from '@angular/core';
 import { Element, ElementType, MainService } from 'src/app/services/main.service';
+import { Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
 
 @Component({
-  selector: 'ui-table',
+  selector: 'app-ui-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss']
 })
-export class TableComponent implements OnInit {
+export class TableComponent implements OnInit, OnDestroy {
+  private readonly toDestroy$ = new Subject<void>();
+
   @Input()
   public set elements(elements: Array<Element>) {
+    if (!elements) { return; }
     /** We want to show same type elements in same column */
     elements.forEach((element) => {
-      element.type = element.type.split('@').shift();
-      this.elementTypeMap[element.type] ? this.elementTypeMap[element.type].push(element) : (this.elementTypeMap[element.type] = [element]);
+      const type = this.takeFirstUriWord(element.type);
+      const found = this.elementTypeMap[type];
+      this.elementTypeMap[type] = [...(found || []), element];
+      this.elementCountPerType[type] = this.elementTypeMap[type].length;
```

Comment:
I think it's as far as the refactor goes, I think this is the shortest I can make. :smile: